File: 02. 구현/#0-1. 상하좌우.swift
import Foundation

let n: Int = Int(readLine()!)!
let plans: [String] = readLine()!.components(separatedBy: " ")

var result: (x: Int, y: Int) = (1, 1)
let moveType: [String] = ["L", "R", "U", "D"]
let moveToXY: [(x: Int, y: Int)] = [(0, -1), (0, 1), (-1, 0), (1, 0)]

for plan in plans {
    var nx = 0, ny = 0
    for i in 0..<moveType.count {
        if plan==moveType[i] {
            nx = result.x + moveToXY[i].x
            ny = result.y + moveToXY[i].y
        }
    }
    
    if nx<1 || ny<1 || nx>n || ny>n {
        continue
    }
    
    result.x = nx
    result.y = ny
}

print("\(result.x) \(result.y)")

File: 02. 구현/#0-1. 상하좌우.py
#0-1. 상하좌우 - p. 110

n = int(input())
plans = list(map(str, input().split()))
x, y = 1, 1

# L, R, U, D에 따른 이동 방향
dx = [0, 0, -1, 1]
dy = [-1, 1, 0, 0]
move_types = ['L', 'R', 'U', 'D']

for plan in plans:
    for i in range(len(move_types)):
        if plan == move_types[i]:
            nx = x + dx[i]
            ny = y + dy[i]
    
    if nx<1 or ny<1 or nx>n or ny>n:
        continue
    
    x, y = nx, ny
        
print(x, y)
